cocos/profiler/counter.ts

```diff
@@ -58,24 +58,24 @@ export class Counter {
         this._accumStart = now;
     }
 
-    public sample (now: number) {
+    public sample (now: number): void {
         this._average(this._value, now);
     }
 
-    public human () {
+    public human (): number {
         const { average, isInteger } = this._opts;
         const v = average ? this._averageValue : this._value;
         return isInteger ? Math.round(v) : Math.round(v * 100) / 100;
     }
 
-    public alarm () {
+    public alarm (): boolean | 0 | undefined {
```

Comment:
I think the return type is boolean only.

Reply:
I don't think there's anything wrong with type inference, it's a matter of the method implementation

the return statement is 
```ts
return (
            (this._opts.below && this._value < this._opts.below)
            || (this._opts.over && this._value > this._opts.over)
        )
```

This means that if `this._opts.below` is 0, it may indeed return 0,
or if `this._opts.below` is undefined, it will return undefined

I would consider changing the return type to `boolean | number | undefined`

In this PR, I don't want to modify the method implementation

Reply:
Interestingly, when we actually start thinking about function return types, we found that the implementation itself has a lot of problems

Reply:
updated https://github.com/cocos/cocos-engine/pull/15314/commits/8843bfe54cd3f8e6f84432948716c8d130ecb233

Reply:
Yep, and please create an issue for it.

Reply:
created https://github.com/cocos/cocos-engine/issues/15355